### PR TITLE
server: clarify documentation around setting and sending headers and ServerStream errors

### PR DESCRIPTION
**Pull request by dfawley**
External documentation: please nit-pick.

RELEASE NOTES:
* server: clarify documentation around setting and sending headers and ServerStream errors

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -1801,12 +1801,26 @@ func (s *Server) getCodec(contentSubtype string) baseCodec {
...
+// the metadata will be sent out when one of the following happens:
+//
+// - grpc.SendHeader is called, or for streaming handlers, stream.SendHeader.
+// - The first response message is sent.  For unary handlers, this occurs when
+//   the handler returns; for streaming handlers, this can happen when stream's
+//   SendMsg method is called.
+// - An RPC status is sent out (error or success).  This occurs when the handler
+//   returns.
+//
+// SetHeader will fail if called after any of these events.
```
Comment (menghanl) [inline server.go:1819]: Make it more clear what `these events` are?
`"the events above"`?

Or move it above, right after the "send out" section

`All the metadata will be sent out when one of the following happens, and SetHeader will fail if called after any of the events:`

File: server.go [hunk 2]
```diff
@@ -1801,12 +1801,26 @@ func (s *Server) getCodec(contentSubtype string) baseCodec {
...
+//
+// The error returned is compatible with the status package.  However, the
+// status will often not match the status seen by the client application, and
```
Comment (menghanl) [inline server.go:1822]: the status __code__ will often not

And did you want to say that the meanings of the codes are different from the client side? Or just that this code might be different from what the client functions return?

**Review by menghanl — CHANGES_REQUESTED** [review]

File: server.go (hunk 1 above)
Comment (dfawley) [reply server.go]: Done (the former)

File: server.go (hunk 2 above)
Comment (dfawley) [reply server.go]: Clarified.  I intended the latter.  The "meaning" of our codes is pretty variable already.

**Review by menghanl — APPROVED** [review]

**Pull request merged**